shared/src/models/utils/position/position-helpers.ts

```diff
@@ -131,3 +132,31 @@ export function simulatedRegionIdOfPosition(position: Position): UUID {
         `Expected position to be in simulatedRegion. Was of type ${position.type}.`
     );
 }
+
+export function upperLeftCornerOf(element: WithExtent): MapCoordinates {
+    const corner = { ...currentCoordinatesOf(element) };
+
+    if (element.size.width < 0) {
+        corner.x += element.size.width;
+    }
+
+    if (element.size.height < 0) {
+        corner.y -= element.size.height;
+    }
+
+    return MapCoordinates.create(corner.x, corner.y);
+}
+
+export function lowerRightCornerOf(element: WithExtent): MapCoordinates {
+    const corner = { ...currentCoordinatesOf(element) };
+
+    if (element.size.width > 0) {
+        corner.x += element.size.width;
+    }
+
+    if (element.size.height > 0) {
+        corner.y -= element.size.height;
+    }
+
+    return MapCoordinates.create(corner.x, corner.y);
+}
```

Comment:
The problem was that position was not always the top-left corner, which could be caused by dragging the lower-right corner to the top-left quadrant of the (previously) top-left corner. At this point, the size would be negative and the position would be the lower-right hand corner.

Reply:
> The problem was that position was not always the top-left corner, which could be caused by dragging the lower-right corner to the top-left quadrant of the (previously) top-left corner. At this point, the size would be negative and the position would be the lower-right hand corner.

So my assumption was correct. Then I probably made an error in the resize interaction, as it would have reassigned the position instead of making the size negative...

Reply:
> > The problem was that position was not always the top-left corner, which could be caused by dragging the lower-right corner to the top-left quadrant of the (previously) top-left corner. At this point, the size would be negative and the position would be the lower-right hand corner.
> 
> So my assumption was correct. Then I probably made an error in the resize interaction, as I thought it would reassign the position instead of making the size negative...

I'm actually not sure that this is an error. I might be wrong, but if i recall correctly, @hpistudent72 and I implemented it this way (with a negative size) when we first made the viewport resize back in April of last year.

Reply:
> Then I probably made an error in the resize interaction, as I thought it would reassign the position instead of making the size negative...

Are you talking about #604? I've just checked out the `main` branch which does not have these commits, but viewports can be resized to negative sizes there, too. I also wanted to check the commit in which the `isInViewport` method was introduced (https://github.com/hpi-sam/digital-fuesim-manv/commit/3904077d2530c0e44a634ebe3f92fc7e32539651), but I did not manage to get the backend running on this commit. However, from looking at the code, I did not see something that would prevent negative sizes.

Therefore, I think the resizing behavior was meant to be like this. We did not find any other places in the code that were using the size of the viewport than the ones I've fixed, so I think its okay as it is now.  
But documenting this behavior is a good idea since the coordinate system is quite confusing (e.g., the global y axis grows upwards, but positive heights grow downwards). I've opened #665 for this.